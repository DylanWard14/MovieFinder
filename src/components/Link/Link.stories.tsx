import React from 'react';

import { ReactComponent as GithubMark } from '../../svgs/github-mark.svg';
import Link from './Link';

export default {
    title: 'Components/Link',
};

export const DefaultExample = () => <Link href="www.google.com">Link text</Link>;

export const WithIconExample = () => (
    <Link href="www.google.com" icon={<GithubMark />}>
        Link text
    </Link>
);
