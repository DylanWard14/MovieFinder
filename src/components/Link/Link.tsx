import React from 'react';
import styled from 'styled-components/macro';

const Component = styled.a`
    color: #ccc;
    fill: #ccc;
    text-decoration: none;
    text-shadow: 1px 1px 1px black;
    font-family: ${props => props.theme.typography.paragraph.fontFamily};
    font-size: 0.9rem;
    display: flex;
    padding: 0.5rem;
    :hover {
        color: ${props => props.theme.colors.main};
        fill: ${props => props.theme.colors.main};
        text-decoration: underline;
    }
    transition: 1s;
`;

const IconWrapper = styled.span`
    width: 1rem;
    display: inline-block;
    padding-right: 0.5rem;
`;

interface LinkProps {
    href: string;
    icon?: React.ReactNode;
}

const Link: React.FunctionComponent<LinkProps> = ({ href, icon, children }) => {
    return (
        <Component href={href} target="__blank">
            <IconWrapper>{icon}</IconWrapper>
            {children}
        </Component>
    );
};

export default Link;
