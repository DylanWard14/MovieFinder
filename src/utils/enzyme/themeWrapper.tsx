import React from 'react';
import { render, shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';

export const renderWrapper = (component: React.ReactNode) => {
    return render(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const shallowWrapper = (component: React.ReactNode) => {
    return shallow(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const mountWrapper = (component: React.ReactNode) => {
    return mount(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};
