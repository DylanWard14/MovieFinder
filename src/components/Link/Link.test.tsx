import React from 'react';
import { render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';
import { ReactComponent as GithubMark } from '../../svgs/github-mark.svg';
import Link from './Link';

// TODO create theme wrapper to auto wrap tests in the ThemeProvider
describe('Link', () => {
    it('renders default correctly', () => {
        const wrapper = render(
            <ThemeProvider theme={myTheme}>
                <Link href="www.google.com" />
            </ThemeProvider>,
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('renders with icon correctly', () => {
        const wrapper = render(
            <ThemeProvider theme={myTheme}>
                <Link href="www.google.com" icon={<GithubMark />} />
            </ThemeProvider>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
