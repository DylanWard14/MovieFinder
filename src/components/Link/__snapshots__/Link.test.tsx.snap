// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders default correctly 1`] = `
<a
  class="Link__Component-di1hpm-0 buMmBA"
  href="www.google.com"
  target="__blank"
>
  <span
    class="Link__IconWrapper-di1hpm-1 zPDDJ"
  />
</a>
`;

exports[`Link renders with icon correctly 1`] = `
<a
  class="Link__Component-di1hpm-0 buMmBA"
  href="www.google.com"
  target="__blank"
>
  <span
    class="Link__IconWrapper-di1hpm-1 zPDDJ"
  >
    <svg>
      github-mark.svg
    </svg>
  </span>
</a>
`;
